src/main/java/bio/terra/service/duos/DuosClient.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.service.duos;
+
+import bio.terra.app.configuration.DuosConfiguration;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.duos.exception.DuosDatasetBadRequestException;
+import bio.terra.service.duos.exception.DuosDatasetNotFoundException;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class DuosClient {
+
+  private final DuosConfiguration duosConfiguration;
+  private final RestTemplate restTemplate;
+
+  @Autowired
+  public DuosClient(DuosConfiguration duosConfiguration, RestTemplate restTemplate) {
+    this.duosConfiguration = duosConfiguration;
+    this.restTemplate = restTemplate;
+    restTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+  }
+
+  /**
+   * @return a new unauthenticated HttpHeaders object for interacting with DUOS APIs
+   */
+  private HttpHeaders getHttpHeaders() {
+    HttpHeaders headers = new HttpHeaders();
+    headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+    return headers;
+  }
+
+  @VisibleForTesting
+  String getDatasetUrl(String duosId) {
+    return String.format("%s/api/tdr/%s", duosConfiguration.getBasePath(), duosId);
+  }
+
+  /**
+   * @param duosId DUOS dataset ID
+   * @param userRequest authenticated user
+   * @return the DuosDataset associated with duosId
+   * @throws DuosDatasetBadRequestException if identifier is malformed
+   * @throws DuosDatasetNotFoundException if no DUOS dataset found
+   */
+  public DuosDataset getDataset(String duosId, AuthenticatedUserRequest userRequest) {
```

Comment:
Note for reviewers: presently we can interact with DUOS as the authed user, as part of facilitating their interaction with our API.

But when we expand the client to fetch a DUOS dataset's authorized users, we will be doing so as a TDR SA:
- TDR SA is the sole admin of the DUOS dataset's Firecloud user group, and the only user who can alter its members
- DUOS Firecloud group syncing may start out as a process manually triggered by a TDR Admin or similar (and thus will have an authed user request associated), but longer term is likely to run as a scheduled background task